src/site/content/en/blog/hid/index.md

```diff
@@ -0,0 +1,478 @@
+  filters: [{ vendorId: 0x05ac, usage: 0x0f, usagePage: 0xff00 }]
+});
+
+// Wait for the HID connection to open.
+await device.open();
+
+// Blink!
+const reportId = 1;
+for (let i = 0; i < 10; i++) {
+  // Turn off
+  await device.sendFeatureReport(reportId, Uint32Array.from([0, 0]));
+  await waitFor(100);
+  // Turn on
+  await device.sendFeatureReport(reportId, Uint32Array.from([512, 0]));
+  await waitFor(100);
+}
+```
+
+{% Glitch { id: 'webhid-apple-keyboard-backlight', path: 'script.js', height: 480, allow: 'hid' } %}
+
+To receive a feature report from a HID device, pass the 8-bit report ID
+associated with the feature report (`reportId`)  to
+`device.receiveFeatureReport()`. The returned promise resolves with a
+[`DataView`] object that contains the contents of the feature report. If the HID
+device does not use report IDs, set `reportId` to 0.
+
+```js
+// Request feature report.
+const dataView = await device.receiveFeatureReport(/* reportId= */ 1);
+
+// Read feature report contents with dataView.getInt8(), getUint8(), etc...
+```
+
+### Listen to connection and disconnection {: #connection-disconnection }
+
+When the website has been granted permission to access a HID device, it can
+actively receive connection and disconnection events by listening to `"connect"`
+and `"disconnect"` events.
+
+```js
+navigator.hid.addEventListener("connect", event => {
+  // TODO: Automatically open event.device or warn user a device is available.
+});
+
+navigator.hid.addEventListener("disconnect", event => {
+  // TODO: Remove |event.device| from the UI.
+});
+```
+
+## Dev Tips {: #dev-tips }
+
+Debugging HID in Chrome is easy with the internal page, `chrome://device-log`
+where you can see all HID and USB device related events in one single place.
+
+<figure class="w-figure">
+  <img src="./device-log-page-screenshot.jpg" class="w-screenshot" alt="Screenshot of the internal page to debug HID">
+  <figcaption class="w-figcaption">Internal page in Chrome to debug HID</figcaption>
+</figure>
+
+## Demos {: #demos }
+
+Some WebHID demos are listed at [web.dev/hid-examples]. Go have a look!
+
+## Security and privacy {: #security-privacy }
+
+The spec authors have designed and implemented the WebHID API using the core
+principles defined in [Controlling Access to Powerful Web Platform Features],
+including user control, transparency, and ergonomics. The ability to use this
+API is primarily gated by a permission model that grants access to only a single
+HID device at a time. In response to a user prompt, the user must take active
+steps to select a particular HID device.
+
+To understand the security tradeoffs, check out the [Security and Privacy
+Considerations] section of the WebHID spec.
+
+On top of this, Chrome inspects the usage of each top-level collection and if a
+top-level collection has a protected usage (.e.g. generic keyboard, mouse), then
+a website won't be able to send to and receive from any reports defined in that
```

Comment:
```suggestion
a website won't be able to send and receive any reports defined in that
```

Reply:
This is not a suggestion to omit needless words. This is a grammatical problem I should have caught before.